core/api/src/app/accounts/get-default-wallet-for-username-phone.ts

```diff
@@ -1,26 +1,67 @@
 import { getWalletFromAccount } from "./get-wallet-from-account"
+import { createInvitedAccountFromPhone } from "./create-account"
 
-import { CouldNotFindWalletFromUsernameAndCurrencyError } from "@/domain/errors"
+import {
+  CouldNotFindUserFromPhoneError,
+  CouldNotFindWalletFromUsernameAndCurrencyError,
+} from "@/domain/errors"
 import { checkedToUsername } from "@/domain/accounts"
 import { checkedToPhoneNumber } from "@/domain/users"
 import { AccountsRepository, UsersRepository } from "@/services/mongoose"
+import { recordExceptionInCurrentSpan } from "@/services/tracing"
+import { ErrorLevel } from "@/domain/shared"
 
 export const getDefaultWalletByUsernameOrPhone = async (
   usernameOrPhone: Username | PhoneNumber,
   walletCurrency?: WalletCurrency,
 ): Promise<Wallet | ApplicationError> => {
-  const checkedUsername = checkedToUsername(usernameOrPhone)
-  if (!(checkedUsername instanceof Error)) {
-    const account = await AccountsRepository().findByUsername(checkedUsername)
-    if (account instanceof Error) return account
-    return getWalletFromAccount(account, walletCurrency)
+  const username = checkedToUsername(usernameOrPhone)
+  if (!(username instanceof Error)) {
+    return getWalletByUsername(username, walletCurrency)
   }
 
-  const checkedPhoneNumber = checkedToPhoneNumber(usernameOrPhone)
-  if (checkedPhoneNumber instanceof Error)
+  const phone = checkedToPhoneNumber(usernameOrPhone)
+  if (phone instanceof Error) {
     return new CouldNotFindWalletFromUsernameAndCurrencyError(usernameOrPhone)
+  }
+
+  return getWalletByPhone(phone, walletCurrency)
+}
+
+const getWalletByUsername = async (
+  username: Username,
+  walletCurrency?: WalletCurrency,
+): Promise<Wallet | ApplicationError> => {
+  const account = await AccountsRepository().findByUsername(username)
+  if (account instanceof Error) return account
+
+  return getWalletFromAccount(account, walletCurrency)
+}
+
+const getWalletByPhone = async (
+  phone: PhoneNumber,
+  walletCurrency?: WalletCurrency,
+): Promise<Wallet | ApplicationError> => {
+  const user = await UsersRepository().findByPhone(phone)
+
+  if (user instanceof CouldNotFindUserFromPhoneError) {
+    const account = await createInvitedAccountFromPhone({ phone })
+    if (account instanceof Error) {
+      recordExceptionInCurrentSpan({
+        error: account,
+        level: ErrorLevel.Warn,
+        attributes: {
+          walletCurrency,
+          phone,
+        },
+      })
```

Comment:
this was with the previous logic, now does not makes sense because getWalletByPhone is returning in a top method (so the exception is recorded)